src/shutdown/umount.c

```diff
@@ -648,6 +648,17 @@ static int umount_with_timeout(MountPoint *m, bool last_try) {
 
                         if (r == -EBUSY && last_try)
                                 log_umount_blockers(m->path);
+
+                        /* If API filesystems under /oldroot cannot be unmounted we can still lazily unmount
+                         * them to unblock /oldroot. They serve no function to us anymore and should be
+                         * memory-only and hence safe to unmount like this. */
+                        if (in_initrd() &&
+                            PATH_STARTSWITH_SET(m->path, "/oldroot/dev", "/oldroot/proc", "/oldroot/sys")) {
+                                log_info("Lazily unmounting '%s' instead.", m->path);
```

Comment:
I don't get this? when would /oldroot ever become visible in the initrd?

Reply:
(btw, we probably should kill oldroot stuff altogether in switch root, as per the upated comments in pivot_root(), which indicate we don't actually need oldroot at all)

Reply:
> I don't get this? when would /oldroot ever become visible in the initrd?

Once you've pivoted back to the initrd during final shutdown? D'uh!

And just to refresh your memory for the hundredth time on this: arch uses systemd-shutdown in its initrd to reduce duplication of effort (well, technically, it creates /run/initramfs as ramfs during shutdown with a service unit).